lib/idempo.rb

```diff
@@ -127,10 +124,15 @@ def response_may_be_persisted?(status, headers, body)
   end
 
   def body_size_within_limit?(response_headers, body)
-    return response_headers["Content-Length"].to_i <= SAVED_RESPONSE_BODY_SIZE_LIMIT if response_headers["Content-Length"]
+    lowercase_response_headers = response_headers.map do |(k, v)|
+      [k.downcase, v]
+    end.to_h
```

Comment:
```suggestion
    lowercase_response_headers = response_headers.to_h do |(k, v)|
      [k.downcase, v]
    end
```

This should work, too. 🤓

Reply:
Or I could look up the rack header hash, don't remember whether it's available in 2.x

Reply:
Rack header hash inherits from Hash, so it's a bit tricky to create it from the standard hash. It also changes the class. And the name of the class is different between Rack 2 and 3. So in the end I've opted to just downcase all the headers.